src/armTemplates/consumption.ts

```diff
@@ -0,0 +1,52 @@
+import { FunctionAppResource } from "./resources/functionApp";
+import { AppInsightsResource } from "./resources/appInsights";
+import { StorageAccountResource } from "./resources/storageAccount";
+import { ArmResourceTemplateGenerator, ArmResourceTemplate } from "../models/armTemplates";
+import { ServerlessAzureConfig } from "../models/serverless";
+
+const resources: ArmResourceTemplateGenerator[] = [
+  FunctionAppResource,
+  AppInsightsResource,
+  StorageAccountResource,
+];
+
+const ConsumptionTemplate: ArmResourceTemplateGenerator = {
```

Comment:
nit: I would call this `ConsumptionPlan`

Reply:
Maybe `ConsumptionPlanTemplate` if changing? I'd hesitate to use the term `ConsumptionPlan` directly

Reply:
It is a template, so will go w/ `ConsumptionPlanTemplate`